girder/utility/s3_assetstore_adapter.py

```diff
@@ -185,7 +252,8 @@ def requestOffset(self, upload):
         return {
             'method': 'PUT',
             'url': url,
-            'headers': headers
+            'headers': headers,
+            'offset': upload['received']
```

Comment:
Yep. This is to accommodate the case of something failing during the first chunk in a proxied upload condition; "dumb" clients expect an `offset` field in the response object, and if it's not there they will not be able to resume an upload unless they choose 0 as the default in that case.

If you would rather I changed it to a literal 0, I am happy to do so.